llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -4053,6 +4053,21 @@ SIRegisterInfo::getNumUsedPhysRegs(const MachineRegisterInfo &MRI,
   return 0;
 }
 
+unsigned
+SIRegisterInfo::getNumDefinedPhysRegs(const MachineRegisterInfo &MRI,
+                                      const TargetRegisterClass &RC) const {
+  auto isDefinedByImplicitDef = [](MachineOperand &Op) {
+    return Op.getParent()->isImplicitDef();
+  };
+
+  for (MCPhysReg Reg : reverse(RC.getRegisters()))
+    for (MCRegAliasIterator AI(Reg, this, true); AI.isValid(); ++AI)
+      if (!(MRI.def_empty(*AI) || std::all_of(MRI.def_begin(*AI), MRI.def_end(),
```

Comment:
def_empty is redundant. Also use all_of(MRI.def_instructions()), you don't care about the specific def operands